larray_editor/arraywidget.py

```diff
@@ -756,19 +755,21 @@ def dropEvent(self, event):
 
     def set_data(self, data=None, bg_value=None):
         # update adapter
-        self.data_adapter.set_data(data, bg_value=bg_value)
-        la_data = self.data_adapter.get_data()
-        axes = la_data.axes
+        if data is None:
+            data = la.LArray([])
+        else:
+            data = la.aslarray(data)
+        axes = data.axes
```

Comment:
IMO, the ArrayEditorWidget should not know what is the type of the data it gets (which is not the case for now...). 
That's why I call `self.data_adapter.set_data` at the beginning of `ArrayEditorWidget.set_data`. 
What I mean is that we will need to be careful once we will be ready to refactor the Adapter (and thus ArrayEditorWidget in the same time).
But for now, current changes are OK.

Reply:
I don't think this is the right way to approach this. ArrayEditorWidget.set_data should either receive an adapter directly or instantiate one depending on the type of data it received. Having a single adapter instance and using set_data on it like we did cannot work because we will need a different adapter class for different data types.

Reply:
Could you add a comment as a reminder?

Reply:
ok

Reply:
Thanks for the comment.